analytics_test.go

```diff
@@ -21,3 +21,57 @@ func TestGeoIPLookup(t *testing.T) {
 		}
 	}
 }
+
+func TestURLReplacer(t *testing.T) {
+	config.AnalyticsConfig.NormaliseUrls.Enabled = true
+	config.AnalyticsConfig.NormaliseUrls.NormaliseUUIDs = true
+	config.AnalyticsConfig.NormaliseUrls.NormaliseNumbers = true
+	config.AnalyticsConfig.NormaliseUrls.Custom = []string{"ihatethisstring"}
+
+	recordUUID1 := AnalyticsRecord{Path: "/15873a748894492162c402d67e92283b/search"}
+	recordUUID2 := AnalyticsRecord{Path: "/CA761232-ED42-11CE-BACD-00AA0057B223/search"}
+	recordUUID3 := AnalyticsRecord{Path: "/ca761232-ed42-11ce-BAcd-00aa0057b223/search"}
+	recordUUID4 := AnalyticsRecord{Path: "/ca761232-ed42-11ce-BAcd-00aa0057b223/search"}
+	recordID1 := AnalyticsRecord{Path: "/widgets/123456/getParams"}
+	recordCust := AnalyticsRecord{Path: "/widgets/123456/getParams/ihatethisstring"}
+
+	config.AnalyticsConfig.NormaliseUrls.compiledPatternSet = InitNormalisationPatterns()
+
+	recordUUID1.NormalisePath()
+	recordUUID2.NormalisePath()
+	recordUUID3.NormalisePath()
+	recordUUID4.NormalisePath()
+	recordID1.NormalisePath()
+	recordCust.NormalisePath()
+
+	if recordUUID1.Path != "/{uuid}/search" {
+		t.Error("Path not altered, is:")
```

Comment:
Sorry, what? If you're asking me why I'm using multiple `Error` calls, this is what the code did. I'm not touching any of the code, just moving it.

This particular anti-pattern is everywhere in both prod and test code, so I'd like to tackle it separately.